### PR TITLE
[ownership] Move ownership passed TempLValueOpt for the stdlib and add an ossa test case.

### PR DESCRIPTION
**Pull request by gottesmm**

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci test

**Comment by gottesmm** [comment]
@swift-ci smoke benchmark

**Comment by gottesmm** [comment]
Not much to change in this pass since it is mainly dealing with projections/destroy_addr/copy_addr

File: test/SILOptimizer/templvalueopt_ossa.sil [hunk 1]
```diff
@@ -0,0 +1,251 @@
+// RUN: %target-sil-opt -enable-sil-verify-all %s -temp-lvalue-opt | %FileCheck %s
+
+import Swift
+import Builtin
+
+// CHECK-LABEL: sil [ossa] @test_enum_with_initialize
+// CHECK:      bb0(%0 : $*Optional<Any>, %1 : $*Any):
```
Comment (meg-gupta) [inline test/SILOptimizer/templvalueopt_ossa.sil:7]: nit : Add : after `// CHECK-LABEL: sil [ossa] @test_enum_with_initialize`
Same in all functions below

**Review by meg-gupta — APPROVED** [review]
LGTM

File: test/SILOptimizer/templvalueopt_ossa.sil (hunk 1 above)
Comment (gottesmm) [reply test/SILOptimizer/templvalueopt_ossa.sil:7]: Thanks! You are right! I should have fixed! I'll fix after the tests pass and then do a smoke test

**Comment by gottesmm** [comment]
Going to fix the nit in a post-commit review.

**Pull request merged**